pkg/gui/controllers/commit_message_controller.go

```diff
@@ -50,30 +55,76 @@ func (self *CommitMessageController) Context() types.Context {
 	return self.context()
 }
 
-// this method is pointless in this context but I'm keeping it consistent
-// with other contexts so that when generics arrive it's easier to refactor
-func (self *CommitMessageController) context() types.Context {
+func (self *CommitMessageController) context() *context.CommitMessageContext {
 	return self.contexts.CommitMessage
 }
 
-func (self *CommitMessageController) confirm() error {
-	message := self.getCommitMessage()
-	self.onCommitAttempt(message)
+func (self *CommitMessageController) handlePreviousCommit() error {
+	// don't increment index if we're on the first commit
+	message, _ := self.helpers.Commits.GetCommitMessageAndDescription()
+	if message == "" && self.context().GetSelectedIndex() != 0 {
```

Comment:
`message` can be empty: `git commit --allow-empty-message`

Reply:
That's a good point, hadn't thought of this edge case. Do you have any suggestions on how we can handle not increasing the index when we are at the first comment of the repo? The only thing I can think of is if we somehow know the total number of commits in the repo we can stop when we reach that number. But I'm not sure if it's even worth handling this case, is it really a big deal if someone reaches the first commit and presses up arrow a few times? Nothing will be broken, it will just show a blank message

Reply:
You can return an identifiable error in `GetCommitMessageFromHistory`  if `formattedHash` is empty,  and handle it.